db/config.cc

```diff
@@ -997,13 +997,14 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , relabel_config_file(this, "relabel_config_file", value_status::Used, "", "Optionally, read relabel config from file")
     , object_storage_config_file(this, "object_storage_config_file", value_status::Used, "", "Optionally, read object-storage endpoints config from file")
     , minimum_keyspace_rf(this, "minimum_keyspace_rf", liveness::LiveUpdate, value_status::Used, 0, "The minimum allowed replication factor when creating or altering a keyspace.")
-    , auth_superuser_name(this, "auth_superuser_name", value_status::Used, "", 
+    , auth_superuser_name(this, "auth_superuser_name", value_status::Used, "",
         "Initial authentication super username. Ignored if authentication tables already contain a super user")
     , auth_superuser_salted_password(this, "auth_superuser_salted_password", value_status::Used, "", 
         "Initial authentication super user salted password. Create using mkpassword or similar. The hashing algorithm used must be available on the node host. "
         "Ignored if authentication tables already contain a super user password.")
     , auth_certificate_role_queries(this, "auth_certificate_role_queries", value_status::Used, { { { "source", "SUBJECT" }, {"query", "CN=([^,]+)" } } },
         "Regular expression used by CertificateAuthenticator to extract role name from an accepted transport authentication certificate subject info.")
+    , live_cql_updates_enabled(this, "live_cql_updates_enabled", liveness::MustRestart, value_status::Used, true, "If set to true, configuration parameters defined with LiveUpdate can be updated in runtime via CQL (by updating system.config virtual table), otherwise they can't.")
```

Comment:
Why MustRestart? It's hostile to operators.

Reply:
The motivation is allowing the DBA to block random users from updating the config for everyone.
These config updates might be dangerous, inconsistent (between nodes), and not persistent.

Another alternative is using RBAC.

Reply:
It's completely unrelated.

Reply:
How so? What stops a user from enabling live_cql_updates_enabled and then updating all other config parameters? can we at least limit live_cql_updates_enabled to REST API update?

Reply:
LiveUpdate means you can change value in scylla.yaml and send SIGHUP instead of a node restart. If the user doesn't have access to scylla.yaml, they can't change anything.

Reply:
Correct, and the main motivation was for cloud not to be able to change config options via system.config table in runtime, but still leave this door opened for on-premise. That's why this param is enabled by default, and will have to be disabled for cloud (set to false in scylla.yaml, once this change is deployed). I assume cloud cannot send signals to reload configuration.

Reply:
The cloud can send signals to reload configuration. Either when an operator does this manually, or one day when they automate it automatically. In any case we need to try to make the system good and free of rough edges (rolling restart = rough edge) and not tailor it to a specific user.